src/autopas/LogicHandler.h

```diff
@@ -1850,20 +1849,26 @@ std::tuple<Configuration, std::unique_ptr<TraversalInterface>, bool> LogicHandle
           }
         });
   } else {
-    if (autoTuner.needsHomogeneityAndMaxDensityBeforePrepare()) {
+    const auto needsDensityStatistics = autoTuner.needsDomainSimilarityStatistics();
+    const auto needsLiveInfo = autoTuner.needsLiveInfo();
+    if (needsDensityStatistics or needsLiveInfo) {
+      // Gather Live Info
       utils::Timer timerCalculateHomogeneity;
       timerCalculateHomogeneity.start();
-      const auto &container = _containerSelector.getCurrentContainer();
-      const auto [homogeneity, maxDensity] = autopas::utils::calculateHomogeneityAndMaxDensity(container);
+      auto particleIter = this->begin(IteratorBehavior::ownedOrHalo);
+      info.gather(particleIter, _neighborListRebuildFrequency, getNumberOfParticlesOwned(), _logicHandlerInfo.boxMin,
+                  _logicHandlerInfo.boxMax, _logicHandlerInfo.cutoff, _logicHandlerInfo.verletSkin);
       timerCalculateHomogeneity.stop();
-      autoTuner.addHomogeneityAndMaxDensity(homogeneity, maxDensity, timerCalculateHomogeneity.getTotalTime());
-    }
-
-    const auto needsLiveInfo = autoTuner.prepareIteration();
-
-    if (needsLiveInfo) {
-      info.gather(_containerSelector.getCurrentContainer(), functor, _neighborListRebuildFrequency);
-      autoTuner.receiveLiveInfo(info);
+      if (needsDensityStatistics) {
+        const auto particleDependentBinDensityStdDev = info.get<double>("particleDependentBinDensityStdDev");
+        const auto particleDependentBinMaxDensity = info.get<double>("particleDependentBinMaxDensity");
+        autoTuner.addDomainSimilarityStatistics(particleDependentBinDensityStdDev, particleDependentBinMaxDensity,
+                                                timerCalculateHomogeneity.getTotalTime());
+        autoTuner.sendDomainSimilarityStatisticsAtStartOfTuningPhase();
+      }
+      if (needsLiveInfo) {
+        autoTuner.receiveLiveInfo(info);
+      }
```

Comment:
If the autotuner needs Density statistics, we anyways gather the full live info, right?
I feel like we could just do
```suggestion
      if (needsDensityStatistics or needsLiveInfo) {
        autoTuner.receiveLiveInfo(info);
      }
```
And then set handle the density stuff in the autotuner instead of having all this logic in here.

Reply:
Nice Idea. Done in c3c7699d. One bit of functionality I changed was that we no longer have a `DEBUG` level log of the cost of getting the domain similarity statistics. But I added a `DEBUG` level log of the cost of gathering live info - which means it can also be used with rule/fuzzy/decision tree tuning, which could be used to determine the cost of getting domain similarity statistics.